packages/data-point/lib/debug-utils/stack.test.js

```diff
@@ -0,0 +1,439 @@
+/* eslint-env jest */
+
+const nock = require('nock')
+const Promise = require('bluebird')
+const DataPoint = require('data-point')
+const { assign, filter, map, find } = DataPoint.helpers
+
+const debugUtils = require('./index')
+const schemaA10 = require('../../test/definitions/schema')
+
+const _true = () => true
+
+const _false = () => false
+
+const identity = input => input
+
+function throwError (acc) {
+  throw new Error('test error message')
+}
+
+function testError (reducer, input) {
+  return dataPoint
+    .resolve(reducer, input)
+    .catch(error => error)
+    .then(result => {
+      expect(result).toBeInstanceOf(Error)
+      expect(result).toHaveProperty('_value')
+      expect(result).toHaveProperty('_stack')
+      expect(result._value).toMatchSnapshot()
+      expect(result._stack).toMatchSnapshot()
```

Comment:
Maybe we want to use `toThrowErrorMatchingSnapshot()`?

Reply:
Does that work with promises?

Reply:
No, it doesn't. ☹️ 

We can do this though, which should write the whole object:

```js
async function testError (reducer, input) {
  await expect(dataPoint.resolve(reducer, input)).rejects.toMatchSnapshot()
}
```
However this changes the snapshots dramatically, and it looks like we lose some critical information stored in the snapshots. So it might not be a good idea to change this after all.

Related thread: https://github.com/facebook/jest/issues/3601